rotkehlchen/tests/unit/accounting/test_misc.py

```diff
@@ -180,3 +185,44 @@ def test_fees_in_received_asset(accountant, google_service):
         AccountingEventType.LEDGER_ACTION: PNL(taxable=FVal('178.615'), free=ZERO),
     })
     check_pnls_and_csv(accountant, expected_pnls, google_service)
+
+
+@pytest.mark.parametrize('mocked_price_queries', [{
+    'ETH': {'GBP': {1609537953: FVal('534.94')}},
+    'USD': {'GBP': {1609537953: FVal('0.741076')}},
+}])
+@pytest.mark.parametrize('should_mock_price_queries', [True])
+def test_main_currency_is_respected(accountant: 'Accountant') -> None:
+    """Verify that the price after processing trades respects the main currency. We change the
+    main currency to GBP and assert the price for a trade using a different currency as quote.
+    """
+    with accountant.db.conn.write_ctx() as cursor:
+        accountant.db.set_setting(cursor, name='main_currency', value=A_GBP)  # type: ignore[arg-type]  # noqa: E501 main_currency is not retrieved manually in any other place of the app
+
+    history: list['AccountingEventMixin'] = [
+        Trade(
+            timestamp=Timestamp(1609537953),
+            location=Location.EXTERNAL,
+            base_asset=A_ETH2,
+            quote_asset=A_USD,
+            trade_type=TradeType.SELL,
+            amount=AssetAmount(FVal('0.04')),
+            rate=Price(FVal('2403.20')),
+            fee=Fee(ZERO),
+            fee_currency=A_EUR,
+            link=None,
+        ),
+    ]
+    accounting_history_process(
+        accountant=accountant,
+        start_ts=Timestamp(1436979735),
+        end_ts=Timestamp(1625001466),
+        history_list=history,
+    )
+    errors = warnings = accountant.msg_aggregator.consume_errors()
+    warnings = accountant.msg_aggregator.consume_warnings()
+    assert len(warnings) == len(errors) == 0
+    # Check that the price is correctly computed in GBP. Since the rate is 2403.20$ the
+    # GBP price should be 2403.20 * 0.741076 == 1780.9538432 being 0.741076 the ratio
+    # of USD to GBP at the moment of the trade
```

Comment:
Don't overdo it with the comments. Mocking stuff also makes these stuff a lot more explicit

```suggestion
    # GBP price should be 2403.20 * 0.741076 == 1780.9538432
```

you can even be a lot more explicit by doing this calculation exactly. By doing `history[0].rate * mocked_price_queries['USD'][GBP'] and get rid of the comment completely